beacon-chain/blockchain/receive_attestation.go

```diff
@@ -37,14 +37,16 @@ type AttestationReceiver interface {
 }
 
 // AttestationTargetState returns the pre state of attestation.
-func (s *Service) AttestationTargetState(ctx context.Context, target *ethpb.Checkpoint) (state.BeaconState, error) {
+func (s *Service) AttestationTargetState(ctx context.Context, target *ethpb.Checkpoint) (state.ReadOnlyBeaconState, error) {
 	ss, err := slots.EpochStart(target.Epoch)
 	if err != nil {
 		return nil, err
 	}
 	if err := slots.ValidateClock(ss, uint64(s.genesisTime.Unix())); err != nil {
 		return nil, err
 	}
+	s.cfg.ForkChoiceStore.RLock()
```

Comment:
Should this be outside of getAttPreState instead of inside?
getAttPreState is used in OnAttestation which don't have the same locks.

Reply:
OnAttestation path is called only from `UpdateHead` that grabs the lock.